### PR TITLE
do not use vagrant shared dir for volumes

### PR DESCRIPTION
**Pull request by pweil-**
Fixes https://github.com/openshift/origin/issues/3435

@jwhonce - now that `empty_dir.go` is trying to chmod volume directories to support arbitrary users running containers we're running into an issue with the volumes being located inside the shared directory.  It manifests itself in the kubelet as:

```
Jun 30 14:26:19 openshift-minion-1 openshift[3090]: E0630 14:26:19.351398    3090 empty_dir.go:322] Expected directory "/vagrant/openshift.local.volumes_openshift-minion-1/pods/f96e07f5-1f33-11e5-94a9-080027c5bfa9/volumes/kubernetes.io~empty-dir/tmp" permissions to be: -rwxrwxrwx; got: -rwxrwxr-x
Jun 30 14:26:19 openshift-minion-1 openshift[3090]: E0630 14:26:19.351767    3090 kubelet.go:1119] Unable to mount volumes for pod "hello-openshift_default": operation not supported; skipping pod
Jun 30 14:26:19 openshift-minion-1 openshift[3090]: E0630 14:26:19.354067    3090 pod_workers.go:108] Error syncing pod f96e07f5-1f33-11e5-94a9-080027c5bfa9, skipping: operation not supported
Jun 30 14:26:29 openshift-minion-1 openshift[3090]: E0630 14:26:29.350041    3090 empty_dir.go:322] Expected directory "/vagrant/openshift.local.volumes_openshift-minion-1/pods/f96e07f5-1f33-11e5-94a9-080027c5bfa9/volumes/kubernetes.io~empty-dir/tmp" permissions to be: -rwxrwxrwx; got: -rwxrwxr-x
Jun 30 14:26:29 openshift-minion-1 openshift[3090]: E0630 14:26:29.350579    3090 kubelet.go:1119] Unable to mount volumes for pod "hello-openshift_default": operation not supported; skipping pod
Jun 30 14:26:29 openshift-minion-1 openshift[3090]: E0630 14:26:29.352949    3090 pod_workers.go:108] Error syncing pod f96e07f5-1f33-11e5-94a9-080027c5bfa9, skipping: operation not supported
```

This change proposes using a volume directory in the root directory.  PTAL

### CODE REVIEW

**Comment by jwhonce** [comment]
@pweil- If you remove ${minion} from the volumes, can  you still create a cluster using vagrant?

**Comment by pweil-** [comment]
@jwhonce since the volume dir will be at the root of each minion's file system and not the vbox shared dir there isn't a need to differentiate in order to avoid the sharing issues we saw before.  Ie. it now lives at `/openshift.local.volumes` just like `/openshift.local.config`

**Comment by jwhonce** [comment]
Then LGTM

File: vagrant/provision-master.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ popd
 echo "Generating certs"
 pushd /vagrant
   SERVER_CONFIG_DIR="`pwd`/openshift.local.config"
-  VOLUMES_DIR="`pwd`/openshift.local.volumes"
+  VOLUMES_DIR="/openshift.local.volumes"
```
Comment (smarterclayton) [inline vagrant/provision-master.sh]: Don't put things in the root - let's find someplace else to stick it.

**Comment by pweil-** [comment]
Just getting back to this.  

@jwhonce - do you have a preferred location to put this if we can't use the root?  Perhaps a `/openshift` directory would be best.  We would probably want to follow up and move the config directories into that directory as well since I believe they live on the root as well (for nodes, when they're copied over).

**Comment by jwhonce** [comment]
@pweil- /openshift sounds good to me. @detiber has moved other config stuff into /etc/openshift as well for the ansible configurations.

**Comment by smarterclayton** [comment]
Why wouldn't they go under /var/lib/openshift?

On Jul 14, 2015, at 10:04 AM, Paul Weil <notifications@github.com> wrote:

Just getting back to this.

@jwhonce <https://github.com/jwhonce> - do you have a preferred location to
put this if we can't use the root? Perhaps a /openshift directory would be
best. We would probably want to follow up and move the config directories
into that directory as well since I believe they live on the root as well
(for nodes, when they're copied over).

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3511#issuecomment-121247806>.

**Comment by jwhonce** [comment]
@detiber Were you the one who pulled them from /var/lib/openshift in the beginning?  I know they were there and then moved to /.

**Comment by pweil-** [comment]
ok, moved this to `/var/lib/openshift`.  @smarterclayton @jwhonce - ptal

**Comment by ncdc** [comment]
1 note re `/var/lib/openshift` - the current SELinux policy sets the directory labels for v2, so you'll potentially get denials on emptyDir and hostDir mounts because the label isn't `svirt_sandbox_file_t`.

**Comment by smarterclayton** [comment]
That's why :)

On Jul 20, 2015, at 8:57 PM, Andy Goldstein <notifications@github.com>
wrote:

1 note re /var/lib/openshift - the current SELinux policy sets the
directory labels for v2, so you'll potentially get denials on emptyDir and
hostDir mounts because the label isn't svirt_sandbox_file_t.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3511#issuecomment-123114408>.

**Comment by jwhonce** [comment]
@ncdc Thanks for the assist!

**Comment by pweil-** [comment]
okie doke, back to `/openshift.local.volumes` unless there are other suggestions. PTAL :-)

**Comment by smarterclayton** [comment]
/var/lib/openshift.local.volumes unless that also has a SELinux policy

On Jul 21, 2015, at 9:37 AM, Paul Weil <notifications@github.com> wrote:

okie doke, back to /openshift.local.volumes unless there are other
suggestions. PTAL :-)

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3511#issuecomment-123306777>.

**Comment by pweil-** [comment]
moved.

```
[vagrant@openshift-minion-1 ~]$ sudo ls -lZ /var/lib
...
drwxr-x---. root root    system_u:object_r:svirt_sandbox_file_t:s0 openshift.local.volumes
...
```

**Comment by jwhonce** [comment]
LGTM

[merge]

**Pull request merged**